File: 202-happy-number/solution.go
func isHappy(n int) bool {
    sVisit := make(map[int]bool)

    for !sVisit[n] {
        sVisit[n] = true
        n = sumOfSquare(n)
        
        if n == 1 {
            return true
        }
    }
    return false
}

func sumOfSquare(n int) int {
    output := 0

    for n>0 {
        digit := n % 10
        digit  = digit * digit
        output += digit
        n = n / 10
    }
    return output
}

File: 202-happy-number/solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        svisit = set()

        while n not in svisit:
            svisit.add(n)
            n = self.sumOfSquares(n)

            if n == 1:
                return True
        return False

    def sumOfSquares(self,n):
        output = 0

        while n:
            digit = n % 10
            digit = digit ** 2
            output += digit
            n = n // 10
        return output